src/processor/connector/vault.py

```diff
@@ -59,3 +62,26 @@ def get_azure_vault_data(secret_key=None):
             val = secret_data['value']
     logger.info('Secret Value: %s', val)
     return val
+
+
+def get_cyberark_data(secret_key=None):
+    """Get secret value for the secret key"""
+    val = None
+    ca_object = config_value('VAULT', 'CA_OBJECT')

```

Comment:
Can `config_value` resolve values from the environment variables?

Reply:
config_value method reads data from the config file. 
Separating the concerns so that each method does one task.
get_client_secret - does get the secret from:
1) connector
2) Env variable.
3) get_vault_data - could be azure vault or cyberark vault. This is in turn calls azure vault fetch or cyberark vault fetch.

Reply:
For azure vault configuration:
[VAULT]
type = azure
tenant_id = <Tenant Id>
client_id = <Service Principal Id>
keyvault = <Keyvault where secrets are stored>

For cyberark vault configuration:
[VAULT]
type = cyberark
CA_OBJECT = <object name>
CA_SAFE = True
CA_EXE = <Path to the cyberark exe>
CA_APPID = <APP ID used for storing the object name>